core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -71,6 +72,9 @@ private[spark] class CoarseGrainedExecutorBackend(
   private[executor] val taskResources = new mutable.HashMap[Long, Map[String, ResourceInformation]]
 
   override def onStart(): Unit = {
+    logInfo("Registering PWR handler.")
+    SignalUtils.register("PWR")(decommissionSelf)
```

Comment:
That's a good point. Right now we register SIGPWR regardless, but we only really need it on K8s & stand alone. Do you think there is any harm in having SIGPWR registered in YARN deployments?

Reply:
I don't know of any specific issues with having it registered on YARN. I was just thinking if not needed no reason to register just in case something unexpected happened.  If we don't need the app config for anything else then at this point I think we can leave it.

Reply:
I think the only situations in which SIGPWR would get sent are from our script or system shutdown. Now if this a graceful system shutdown which didn't fall back to an unblockable signal that might be annoying, but I think on servers it should be fine. But lets leave this comment here incase anyone else has a counter point.